sklearn/metrics/ranking.py

```diff
@@ -88,8 +88,8 @@ def auc(x, y, reorder=False):
         x, y = x[order], y[order]
     else:
         dx = np.diff(x)
-        if np.any(dx < 0):
-            if np.all(dx <= 0):
+        if np.any(dx < -1e-10):
```

Comment:
Is this change needed?

Reply:
@lesteve Thanks. 
I think we need it. Because there are cases when x is something like [0.8, 0.8+1e-12, 0.8], For such case, we shouldn't raise an error here.